students/Lincoln_Zhang/Session2/grid_print.py

```diff
@@ -21,22 +21,28 @@
 
 boxsize = int(input("Please input box size: "))
 gridsize = int(input("Please input grid size: "))
+n = int(input("Please input a positive integer: ")) + 1
 
 def horizontal_line(boxsize):
 	return "- " * boxsize + "+ "
 
 def vertical_line(boxsize):
 	return "  " * boxsize + "| " 
 
-# def grid_print_1(boxsize, gridsize):
-# 	r = horizontal_line(boxsize)
-# 	c = vertical_line(boxsize)
-# 	for _ in range(gridsize+1):
-# 		if _%boxsize == 0:
-# 			print ("+ " + r*gridsize)
-# 		else:
-# 			for _ in range(boxsize):
-# 				print ("| " + c*gridsize)
+#a scale function for a 4X4 grid
+def grid_print_1(n):
```

Comment:
I'm too sure what this is doing. We will not need to sanitize any input to make sure it is modulo 4. 

Refer to the first example shown: https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/grid_printer.html

At the very top, we print a 4x4 grid with grid_print(9). The 9 here represents the total length of 1 side. If you count the number of '-' characters (and the '+') in the middle, we see a total of 9. This is what the parameter is responsible for. Similarly, follow the grid_print(3) and grid_print(15) examples, they still create a 4x4 grid but the sides are of length 3 and 15 respectively.

Reply:
Thanks I see what you are saying. check out this revision please.